backend/api/README.md

```diff
@@ -9,23 +9,35 @@ Tools needed:
 
 ## Auto-generation of Go client and swagger definitions
 
+Set environment variable `API_VERSION` to the version that you want to generate. We use `v1beta1` as example here.
+
+```bash
+API_VERSION="v1beta1" make generate
+```
+
+For local developments, use 
 ```bash
-make generate
+API_VERSION="v1beta1" make generate-from-scratch
```

Comment:
remove `API_VERSION="v1beta1"` as it's defined already from above.

Reply:
Also why change `make generate` to `make generate-from-scratch`?

Reply:
I think the wording here is a bit confusing. There are two ways to generate the api package: using `make generate`, which uses a pre-built api-generator image; or you can use `make generate-from-scratch`, which starts by build the api-generator image locally. I updated the wording to make it clearer.

Reply:
And `API_VERSION="v1beta1"` is an integral part of the command.

Reply:
I don't get why you need to build the image for local development, you're not changing anything related to the image. Did you hit any specific error with `make generate` that leads you to rebuilding the image? If the hosted image is broken, we need to update it.

All the nitpicks combined was to simply reuse `API_VERSION` across the doc. 
The doc would be something like this:
## Set version
```bash
export API_VERSION="v1beta1"
```
## Generate go client
```bash
make generate
```
Code will be generated in
* `./"${API_VERSION}"/go_client`
* ...
## Generate python client
```bash
./build_kfp_server_api_python_package.sh
```
Code will be generated in `./"${API_VERSION}"/python_http_client`.

(need changes to `build_kfp_server_api_python_package.sh` so it reads `$API_VERSION`)

Reply:
In this PR the image is not changed, but in future developments, if the Dockerfile is changed, the image need to be rebuilt. This is why there are two options to generate the packages.

Reply:
As to set version as an environment variable, I will update the doc accordingly.

Reply:
The command `make-generate-from-scratch` has always been in the `Makefile`. I just added more lines in the readme so it's more convenient for future devs.